### PR TITLE
Widget tracking - Fingerprint to UUID

### PR DESCRIPTION
**Pull request by sanbornsen**

### CODE REVIEW

File: sezzle.js [hunk 1]
```diff
@@ -1115,34 +1118,43 @@ SezzleJS.prototype.replaceBanner = function () {
...
-      'cart_id': this.getCookie('cart'),
-      'fingerprint': this.fingerprint,
-      'ip_address': this.ip,
-      'merchant_site': window.location.hostname,
-      'is_mobile_browser': this.isMobileBrowser(),
-      'user_agent': navigator.userAgent,
-      'merchant_uuid': this.merchantID,
-    }));
+  this.track_id = this.getTrackId();
+  let viewport = {
```
Comment (sanbornsen) [inline sezzle.js]: We can't use `let` because it's an ES6 feature. Doesn't support in older versions.

File: initial-script.js [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+  document.body.appendChild(iframe);
+  callback(track_id);
 
-(function() {
-  var uuid = getUUID();
+})(function (track_id) {
+  console.log('track id', track_id);
   var el = document.createElement('script');
-  el.src = 'https://widget.sezzle.com/v1/javascript/price-widget?uuid=' + uuid
+  el.src = 'http://localhost:12121/v1/javascript/price-widget?uuid=' + document.sezzleConfig['merchantID'] + '&track_id=' + track_id
```
Comment (sanbornsen) [inline initial-script.js]: Don't forget to change the URL

File: initial-script.js [hunk 3]
```diff
@@ -1,15 +1,40 @@
...
+  iframe.style.visibility = 'hidden';
+  let html = '<body>Foor</body>';
+  iframe.src = 'https://staging.tracking.sezzle.com?trk_id=' + track_id;
+  document.body.appendChild(iframe);
+  callback(track_id);
 
-(function() {
-  var uuid = getUUID();
+})(function (track_id) {
+  console.log('track id', track_id);
```
Comment (sanbornsen) [inline initial-script.js]: Don't forget to remove the unwanted `console.log`s

File: initial.js [hunk 4]
```diff
@@ -1,72 +1,26 @@
...
+(function(callback) {
+  let track_id = null;
+  function generateUUID() {
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+      var r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);
+      return v.toString(16);
     });
+  }
+  function getCookie(name) {
+    var value = "; " + document.cookie;
```
Comment (sanbornsen) [inline initial.js]: Use `'` instead of `"`

File: initial.js [hunk 5]
```diff
@@ -1,72 +1,26 @@
...
+    var parts = value.split("; " + name + "=");
```
Comment (sanbornsen) [inline initial.js]: Use `'` instead of `"`

File: initial.js [hunk 6]
```diff
@@ -1,72 +1,26 @@
...
+    if (parts.length === 2) {
+      return parts.pop().split(";").shift();
```
Comment (sanbornsen) [inline initial.js]: Use `'` instead of `"`

File: initial.js [hunk 7]
```diff
@@ -1,72 +1,26 @@
...
-        var el = document.createElement('script');
-        el.src = 'https://widget.sezzle.com/v1/javascript/price-widget?uuid=' + document.sezzleConfig["merchantID"] + '&fingerprint=' + fingerprint
-        document.getElementsByTagName('head')[0].appendChild(el);
-    });
-}
\ No newline at end of file
+  track_id = getCookie("szl_uuid")
+  if (!track_id) {
+    track_id = generateUUID();
+    document.cookie = "szl_uuid=" + track_id + ";domain=.sezzle.com;path=/";
```
Comment (sanbornsen) [inline initial.js]: Use `'` instead of `"`

File: initial-script.js (hunk 2 above)
Comment (rajan-bhuyan-sezzle) [reply initial-script.js]: I have kept it as `localhost` as of now. Will change them when we are done fully

**Pull request merged**